### PR TITLE
Run workflows on all OS versions in GHA.

### LINKED ISSUE #4610: [Proposal] Strictly define OS version in GHA
## Is your feature request related to a problem? Please describe.
Now all GHA workflows use `ubuntu-latest` runners. The version of OS might be changed in future which might cause failures or crashes.
See example:
https://github.com/opensearch-project/OpenSearch/blob/9b2202aeb3a881c0c2ad72522fa1e2656d5d4d1e/.github/workflows/gradle-check.yml#L13

## Describe the solution you'd like
I propose to strictly define the OS version which should be used there. Currently `ubuntu-latest` = `ubuntu-20.04` ([link](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)). The proposed fix should change all references to `ubuntu-latest` by `ubuntu-20.04`.

## Describe alternatives you've considered
Do nothing until GH publish OS version change for `ubuntu-latest`. There will be a few month period to update and fix all workflows if needed.

## Additional context
**Tags**: GHA, CI, ubuntu, ubuntu-latest, runner, OS version

I can to the fix (a PR) on my own, but I need an approval to that idea first.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
### Description
Update some CI workflows to run on all available OS versions.
This could be an overkill, but it helps to early detect CI OS templates deprecation or breaking change.

### Issues Resolved
Fixes #4610

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/changelog_verifier.yml [hunk 1]
```diff
@@ -6,7 +6,10 @@ on:
 jobs:
   # Enforces the update of a changelog file on every pull request
   verify-changelog:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ 'ubuntu-22.04', 'ubuntu-20.04', 'ubuntu-latest' ]
```
Comment (owaiskazi19) [inline .github/workflows/changelog_verifier.yml:12]: Thanks for the PR @Yury-Fridlyand! Just wanted to understand is there any specific reason for picking up `22.04` and `20.04` version of ubuntu?

File: .github/workflows/gradle-check.yml [hunk 2]
```diff
@@ -16,8 +16,10 @@ jobs:
     permissions:
       contents: read # to fetch code (actions/checkout)
       pull-requests: write # to create or update comment (peter-evans/create-or-update-comment)
-
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```
Comment (saratvemulapalli) [inline .github/workflows/gradle-check.yml:19]: Gradle check workflow kicks off a run on Jenkins which is running on Linux.

**Review by saratvemulapalli — COMMENTED** [review]
Bunch of workflows here are to maintain external sanity but they dont add value running on different machines like link checker, code hygene etc.

**Comment by reta** [comment]
To be fair, I think it is overkill to run in different Ubuntu version, I would see the benefits going with different distributions (Debian, CentOS, ...) though

**Comment by Yury-Fridlyand** [comment]
> To be fair, I think it is overkill to run on different Ubuntu versions

I agree.

This PR's goal to prove that all CI action properly could run on 20.04 and on 22.04 and it is safe to replace `ubuntu-latest` by a specific version. Having version strictly defined we won't depend on sudden `latest` version bump.

> I would see the benefits going with different distributions (Debian, CentOS, ...) though

Tests are executed in Jenkins and its config isn't completely open[source], so I can't propose or test any changes there.

**Comment by dblock** [comment]
Want to finish this PR along those lines @Yury-Fridlyand (run only on 1 ubuntu?).

Btw, the Jenkins CI *is* completely open-source, but may be a bit difficult to find where all those things are.

**Comment by reta** [comment]
> Want to finish this PR along those lines @Yury-Fridlyand (run only on 1 ubuntu?).

We already run on `ubuntu-latest`

> This PR's goal to prove that all CI action properly could run on 20.04 and on 22.04 and it is safe to replace `ubuntu-latest` by a specific version. Having version strictly defined we won't depend on sudden latest version bump.

I think replacing `ubuntu-latest` by specific version would cut us off from new releases (I doubt someone will remember to go and update Github actions across all branches to use fe upcoming 23.04, ...). 

May be you could share the example of the issue you run into so you came to the conclusion that versions should be defined? AFAIK, we don't use anything OS specific (besides `docker-compose`) and rely on `Gradle` wrapper to do the job of building the project. And `Gradle` does make sure the wrappers work across the matrix of OSes.

**Comment by Yury-Fridlyand** [comment]
When I started working on this change `ubuntu-latest` was 20.04. The goal was to check that we can switch to newer version without having troubles.
Currently, `ubuntu-latest` is 22.04 and there is nothing to do left. I hope the next runner upgrade will be smooth too.

**Pull request closed without merging**